modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala

```diff
@@ -0,0 +1,156 @@
+package org.tessellation.infrastructure.snapshot
+
+import cats.data.NonEmptyList
+import cats.effect.Async
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.functorFilter._
+import cats.syntax.list._
+import cats.syntax.option._
+import cats.syntax.order._
+import cats.{Eval, MonadThrow}
+
+import org.tessellation.dag.snapshot.{GlobalSnapshot, GlobalSnapshotInfo, StateChannelSnapshotBinary}
+import org.tessellation.domain.snapshot.{GlobalSnapshotStorage, TimeSnapshotTrigger, TipSnapshotTrigger}
+import org.tessellation.ext.cats.syntax.next._
+import org.tessellation.ext.crypto._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.height.SubHeight
+import org.tessellation.sdk.domain.consensus.ConsensusFunctions
+import org.tessellation.security.hash.Hash
+import org.tessellation.security.signature.Signed
+
+import eu.timepit.refined.types.numeric.NonNegLong
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import shapeless.Typeable
+
+trait GlobalSnapshotConsensusFunctions[F[_]]
+    extends ConsensusFunctions[F, GlobalSnapshotEvent, GlobalSnapshotKey, GlobalSnapshotArtifact] {}
+
+object GlobalSnapshotConsensusFunctions {
+
+  def make[F[_]: Async: KryoSerializer: MonadThrow](
+    globalSnapshotStorage: GlobalSnapshotStorage[F],
+    heightInterval: NonNegLong
+  ): GlobalSnapshotConsensusFunctions[F] = new GlobalSnapshotConsensusFunctions[F] {
+
+    private val logger = Slf4jLogger.getLoggerFromClass(GlobalSnapshotConsensusFunctions.getClass)
+    private val tipSnapshotTriggerTypable = Typeable[TipSnapshotTrigger]
+
+    def consumeSignedMajorityArtifact(signedArtifact: Signed[GlobalSnapshotArtifact]): F[Unit] =
+      globalSnapshotStorage.save(signedArtifact)
+
+    def triggerPredicate(
+      last: (GlobalSnapshotKey, GlobalSnapshotArtifact),
+      event: GlobalSnapshotEvent
+    ): Boolean = event.toOption.flatMap(_.toOption).fold(false) {
+      case TipSnapshotTrigger(height) => last._2.height.nextN(heightInterval) === height
```

Comment:
to use `.eqv`I think you need to import `syntax.eq._` 🤔

Reply:
`syntax.order._` already covers stuff in `syntax.eq._`